docs/data/data-grid/rows/rows.md

```diff
@@ -279,28 +279,71 @@ For now, row reordering is disabled if sorting is applied to the grid.
 In addition, if row grouping or tree data is being used, the row reordering is also disabled.
 :::
 
-## 🚧 Row spanning
+## Row pinning [<span class="plan-pro"></span>](https://mui.com/store/items/material-ui-pro/)
+
+Pinned (or frozen, locked or floating) rows are rows that are visible at all times while the user scrolls the grid vertically.
+Pinned rows are not affected by sorting or filtering.
 
 :::warning
-This feature isn't implemented yet. It's coming.
+This feature is experimental, it needs to be explicitly activated using the `rowPinning` experimental feature flag.
+
+```tsx
+<DataGridPro experimentalFeatures={{ rowPinning: true }} {...otherProps} />
+```
 
-👍 Upvote [issue #207](https://github.com/mui/mui-x/issues/207) if you want to see it land faster.
 :::
 
-Each cell takes up the width of one row.
-Row spanning allows to change this default behavior.
-It allows cells to span multiple rows.
-This is very close to the "row spanning" in an HTML `<table>`.
+You can pin rows at the top or bottom of the grid by passing pinned rows data through the `pinnedRows` prop:
+
+```tsx
+const pinnedRows: GridPinnedRowsProp = {
+  top: [{ id: 0, brand: 'Nike' }],
+  bottom: [
+    { id: 1, brand: 'Adidas' },
+    { id: 2, brand: 'Puma' },
+  ],
+};
+
+<DataGridPro pinnedRows={pinnedRows} />;
+```
+
+:::warning
+Just like the `rows` prop, `pinnedRows` prop should keep the same reference between two renders.
+Otherwise, the grid will re-apply heavy work like sorting and filtering.
+:::
+
+The data format for pinned rows is the same as for the `rows` prop (see [Feeding data](/x/react-data-grid/rows/#feeding-data)).
+
+Pinned rows data should also meet [Row identifier](/x/react-data-grid/rows/#row-identifier) requirements.
+
+{{"demo": "RowPinning.js", "disableAd": true, "bg": "inline"}}
+
+### Controlling pinned rows
```

Comment:
I know "controlling" is common wording to document this kind of ability.

But I feel this particular section could be titled differently because it's likely going to be one of the most popular use cases of row pinning (allowing users to pin their own rows). 
Perhaps we want to facilitate the search on it.

How about

```suggestion
### Row pinning for end-users (controlling state)
```

?

Reply:
Not sure about the naming.
This demo demonstrates how developer can control pinned rows. It also empowers the end user to pin/unpin rows using UI, but it's just one of the use-cases.

Reply:
Alright. We can see how this use case will emerge.